Compilation error on gcc 3.4.6

Hello, 

I hit some trouble to use last version on old RedHat:

In file included from JavaScriptRuntime.cc:126:
/home/albert/albert/3albert/include/arch_independent/rapidjson/document.h: In destructor `rapidjson::GenericValue<Encoding, Allocator>::~GenericValue()':
/home/albert/albert/3albert/include/arch_independent/rapidjson/document.h:489: error: expected class-name before '(' token
/home/albert/albert/3albert/include/arch_independent/rapidjson/document.h: In member function`typename rapidjson::GenericMemberIterator<false, Encoding, Allocator>::Iterator rapidjson::GenericValue<Encoding, Allocator>::RemoveMember(typename rapidjson::GenericMemberIterator< false, Encoding, Allocator>::Iterator)':
/home/albert/albert/3albert/include/arch_independent/rapidjson/document.h:924: error: expected class-name before '(' token

Error come from:
  m->~GenericMember();

gcc -v 
Lecture des spécification à partir de /usr/lib/gcc/x86_64-redhat-linux/3.4.6/specs
Configuré avec: ../configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --enable-shared --enable-threads=posix --disable-checking --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-java-awt=gtk --host=x86_64-redhat-linux
Modèle de thread: posix
version gcc 3.4.6 20060404 (Red Hat 3.4.6-9)

Of course build without problem on the same host with gcc-4.4.7

Any clue ?
Thanks

--- a/include/rapidjson/document.h
+++ b/include/rapidjson/document.h
@@ -535,7 +535,7 @@ public:
 
             case kObjectFlag:
                 for (MemberIterator m = MemberBegin(); m != MemberEnd(); ++m) {
-                    m->~GenericMember();
+                    m->~Member();
                 }
                 Allocator::Free(data_.o.members);
                 break;
@@ -977,7 +977,7 @@ public:
         }
         else {
             // Only one left, just destroy
-            m->~GenericMember();
+            m->~Member();
         }
         --data_.o.size;
         return m;

